### ISSUE TITLE
Split `elements` into `elements` and `elements-core`

### ISSUE THREAD

**Issue by marcelltoth** [stoplightio/elements#1048]
*Epic: #996*

## Summary

As described in the epic, extract another package called `elements-core` that holds the common material from `API` and `StoplightProject`. Also split the storybook -> move any internal component demos to `@stoplight/elements-core`'s storybook instance (which you should create).

### Out of scope

Creating `elements-dev-portal`. For this issue, keep both `API` and `StoplightProject` in `elements`.

## Additional information

This is ideally done after #1047 so you don't have to deal with setting up publishing for the new package, but can be started earlier to better parallelize work.

**Comment by marcelltoth** [comment]
@mallachari I don't think this should be closed yet, correct?

**Comment by philsturgeon** [comment]
We can close this now as we have elements-core published as a beta.
